kartothek/utils/migration_helpers.py

```diff
@@ -0,0 +1,196 @@
+import inspect
+import traceback
+import warnings
+from functools import partial, wraps
+from typing import Callable, Tuple
+
+multi_table_feature_deprecation_warning = (
+    "The `{parameter}` keyword is deprecated and will be removed in the next major"
+    "release in an effort to remove the multi table feature."
+)
```

Comment:
Do we use this variable somewhere?

Reply:
This variable is supposed to be passed as first decorator parameter for the current migration effort.

It is not very clean without use right now, but I think it is not located too badly once in use.

Reply:
Ah, ok. I think we could call it `MULTI_TABLE_FEATURE_DEPRECATION_WARNING` to highlight that it is a global constant. Or maybe even better, create a small utility:
```python
def warning_message(parameter: str) -> str:
    return MULTI_TABLE_FEATURE_DEPRECATION_WARNING.format(parameter=parameter)
```

Reply:
I think having a function like that could better communicate the rationale why this variable exists to a user.

Reply:
I will however capitalize it as constant.

Reply:
Sidenote because I did not address the second option: The proposed small utility is unsuitable because the decorator has to keep the authority on what parameter to raise a warning with and this selection is dynamic, based on the individual function calls for the deprecate_parameters_if_set decorator.